Update the Godot Android Editor name from `Godot Editor 4.x` to `Godot Editor 4`

m4gr3d opened the pull request:
<!--
Please target the `master` branch in priority.
PRs can target `3.x` if the same change was done in `master`, or is not relevant there.

Relevant fixes are cherry-picked for stable branches as needed by maintainers.
You can mention in the description if the change is compatible with `3.x`.
-->

Commit:
Update the Godot Android Editor name from `Godot Editor 4.x` to `Godot Editor v4`

akien-mga commented:
We don't really use "v4" anywhere to brand Godot. It's either "Godot Engine", "Godot Engine 4.0", "Godot Engine 4" or "Godot Engine 4.x" (the "4" one is also not used much, to avoid the similarity with Unreal Engine 4, but if we want to keep it generic while different from Godot 3 here, it could work).

m4gr3d commented:
> We don't really use "v4" anywhere to brand Godot. It's either "Godot Engine", "Godot Engine 4.0", "Godot Engine 4" or "Godot Engine 4.x" (the "4" one is also not used much, to avoid the similarity with Unreal Engine 4, but if we want to keep it generic while different from Godot 3 here, it could work).

Updated to `Godot Editor 4`, and updated the play store label to match as well.

akien-mga commented:
Thanks!
